src/vos/vos_io.c

```diff
@@ -1769,23 +1811,22 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 		goto abort;
 	}
 
-	/* Publish NVMe reservations */
-	err = vos_publish_blocks(ioc->ic_cont, &ioc->ic_blk_exts, true,
-				 VOS_IOS_GENERIC);
-
-	if (dth != NULL && err == 0)
-		err = vos_dtx_prepared(dth);
+	/* Publish SCM reservations */
+	err = vos_publish_scm(ioc->ic_cont, &ioc->ic_rsrvd_scm, true, dth);
+	if (err == 0)
+		/* Publish NVMe reservations */
+		err = vos_publish_blocks(ioc->ic_cont, &ioc->ic_blk_exts, true,
+					 VOS_IOS_GENERIC, dth);
```

Comment:
Why don't we do the dtx loigc here?
- if dth == NULL, do real publish;
- if dth != NULL, collect reserved list for the real pubish in vos_tx_end()?

Reply:
Yes, we did as you expected, the vos_publish_scm()/vos_publish_blocks() will handle that internally: if the "dth" is NULL, it will do the real publish.